### ISSUE TITLE
Menu items with missing localization resources cause all menu items to fail to localize

### ISSUE THREAD

**Issue by lyonsil** [paranext/paranext-core#825]
**Describe the bug**
Tom showed me an extension with a contributed menu JSON file. That menu JSON referenced a localization key that wasn't included in the localized resources. All menu items then failed to localize instead of just the ones with missing keys.

**Expected behavior**
All menu items that include localized keys should render properly instead of them all failing.

**Comment by tjcouch-sil** [comment]
I have also found this. It is quite annoying. However, I believe this is possibly by design. We decided that requesting a localized string for a key that doesn't have a value should throw in development, and I believe that's what's happening here. In production, we return the localized string key and log a warning to my understanding. Maybe we should consider revising this design overall or making it very obvious in some other way what's going wrong (maybe we can implement toasts? Or renderer error log?).
